pytorch_lightning/callbacks/progress/tqdm_progress.py

```diff
@@ -216,87 +225,74 @@ def on_sanity_check_end(self, trainer, pl_module):
         self.val_progress_bar.close()
 
     def on_train_start(self, trainer, pl_module):
-        super().on_train_start(trainer, pl_module)
         self.main_progress_bar = self.init_train_tqdm()
 
     def on_train_epoch_start(self, trainer, pl_module):
-        super().on_train_epoch_start(trainer, pl_module)
         total_train_batches = self.total_train_batches
         total_val_batches = self.total_val_batches
         if total_train_batches != float("inf") and total_val_batches != float("inf"):
             # val can be checked multiple times per epoch
             val_checks_per_epoch = total_train_batches // trainer.val_check_batch
             total_val_batches = total_val_batches * val_checks_per_epoch
         total_batches = total_train_batches + total_val_batches
-        reset(self.main_progress_bar, total=total_batches, current=self.train_batch_idx)
+        self.main_progress_bar.total = convert_inf(total_batches)
         self.main_progress_bar.set_description(f"Epoch {trainer.current_epoch}")
 
     def on_train_batch_end(self, trainer, pl_module, outputs, batch, batch_idx):
-        super().on_train_batch_end(trainer, pl_module, outputs, batch, batch_idx)
         if self._should_update(self.train_batch_idx):
-            self._update_bar(self.main_progress_bar)
+            _update_n(self.main_progress_bar, self.train_batch_idx + self._val_processed)
             self.main_progress_bar.set_postfix(self.get_metrics(trainer, pl_module))
 
     def on_train_epoch_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
-        if self.is_enabled:
-            self._update_bar(self.main_progress_bar)
+        _update_n(self.main_progress_bar, self.train_batch_idx + self._val_processed)
```

Comment:
Why updating if the progress bar can disabled ?

Reply:
`_update_n` checks for disabled